libkernelflinger/vars.c

```diff
@@ -613,7 +613,9 @@ static void CDD_clean_string(char *buf)
 	if (!buffer[0]) { \
 		UINTN bufsz = sizeof(buffer); \
 		char *dmidata = SMBIOS_GET_STRING(type, field); \
-		if (dmidata && dmidata != SMBIOS_UNDEFINED) { \
+		if (dmidata && dmidata != SMBIOS_UNDEFINED && \
+				dmidata[0] != '\0' && \
+				!(dmidata[0] == ' ' && dmidata[1] == '\0')) { \
```

Comment:
You could create a function like:

    static BOOLEAN valid_dmidata(char *data)
    {
    	return dmidata && dmidata != SMBIOS_UNDEFINED &&
    		*dmidata && !(dmidata[0] == ' ' && dmidata[1] != '\0');
    }

And this change would become:

    -		if (dmidata && dmidata != SMBIOS_UNDEFINED) { \
    +		if (valid_dmidata(dmidata)) { \